File: src/main/java/spp/example/webapp/edge/TwoMbObject.java
package spp.example.webapp.edge;

import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;
import java.io.OutputStream;
import java.io.PrintStream;
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

@Component
public class TwoMbObject implements Runnable {

    private final PrintStream nullStream = new PrintStream(new OutputStream() {
        public void write(int b) {
        }
    });

    private final byte[] TWO_MB_ARR = new byte[1024 * 1024 * 2];

    @PostConstruct
    public void init() {
        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
        executorService.scheduleAtFixedRate(this, 0, 1, TimeUnit.SECONDS);
    }

    @Override
    public void run() {
        nullStream.println(TWO_MB_ARR);
        nullStream.print("nothing");
    }
}

File: src/main/kotlin/example/webapp/edge/TwoMbObject.kt
package example.webapp.edge

import org.springframework.stereotype.Component
import java.io.OutputStream
import java.io.PrintStream
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit
import javax.annotation.PostConstruct

@Component
class TwoMbObject : Runnable {
    private val nullStream = PrintStream(object : OutputStream() {
        override fun write(b: Int) {}
    })
    private val TWO_MB_ARR = ByteArray(1024 * 1024 * 2)
    @PostConstruct
    fun init() {
        val executorService = Executors.newSingleThreadScheduledExecutor()
        executorService.scheduleAtFixedRate(this, 0, 1, TimeUnit.SECONDS)
    }

    override fun run() {
        nullStream.println(TWO_MB_ARR)
        nullStream.print("nothing")
    }
}